assets/scripts/segments/Segment.jsx

```diff
@@ -214,15 +214,15 @@ export class Segment extends React.Component {
 
         // If the shift key is pressed, we remove all segments
         if (event.shiftKey === true) {
-          this.props.clearSegments()
+          this.props.clearSegmentsAction()
           infoBubble.hide()
-          this.props.showStatusMessage(t('toast.all-segments-deleted', 'All segments have been removed.'))
+          this.props.showStatusMessage(t('toast.all-segments-deleted', 'All segments have been removed.'), true)
```

Comment:
🤔 I'm not sure it's intuitive to expect calls from the component to know the details of how they need to dispatch actions. Maybe `incrementSegmentWidth` and `showStatusMessage` should still use the explicit declaration with dispatch? But I would love to hear what other people think.

Reply:
I found in description of issue that one way to refactor is move arguments inside a component so that why I put it there)
Nevertheless if I would better to leave previous solution I will correct it accourdingly)

Reply:
This is an instance where I'm inclined to accept it the way it's done here, for the following reasons:

1. `showStatusMessage` will be refactored in the near future anyway. 
2. `incrementSegmentWidth` seems to have an overly complicated signature anyway. Why does a increment action need to know the original width of the segment? (The short answer is that this is a relic of an older function signature that existed before we implemented Redux). It's one thing to hide the implementation but I think what it really surfaces for me here is the need to revisit how this is implemented.
3. Since the rest of the prop dispatches are simple, there's no need to make them all more complex for the sake of these previous two.

Reply:
Another way to say my comment is that this is an opportunity to move some of the complexity of the action creators into the body of the action creator itself. That way neither the component or the connect function of this or any other component needs to be responsible for it. But this is outside of the scope of this PR.

I'm glad this was raised, it helps us look at targets for future improvement!